zeth-contracts/contracts/MerkleTreeMiMC7.sol

```diff
@@ -7,29 +7,110 @@ pragma solidity ^0.5.0;
 import "./BaseMerkleTree.sol";
 import "./MiMC7.sol";
 
-contract MerkleTreeMiMC7 is BaseMerkleTree {
+// The Merkle tree implementation must trade-off complexity, storage,
+// initialization cost, and update & root computation cost.
+//
+// This implementation stores all leaves and nodes, skipping those that have not
+// been populated yet.  The final entry in each row stores the default value for
+// that row.
+contract MerkleTreeMiMC7 is BaseMerkleTree
+{
+    uint256 constant MASK_LS_BIT = uint(~1);
```

Comment:
What is the `~` here?

Reply:
It's the not operator.  I unset the least-significant bit using this.

Reply:
Oh I see.
I was confused by the operator precedence (see: https://solidity.readthedocs.io/en/v0.6.2/miscellaneous.html#order-of-precedence-of-operators)
What about writing `uint256 constant MASK_LS_BIT = ~uint256(1);` instead?

Reply:
changed

Reply:
Thanks